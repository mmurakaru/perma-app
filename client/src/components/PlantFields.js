import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import logo from '../assets/perma_logo.svg'
import { ReactComponent as ArrowIcon } from '../assets/arrow_down.svg'
import { Link } from 'react-router-dom'

PlantFields.propTypes = {
    plant: PropTypes.object,
    isLoaded: PropTypes.bool,
    backToOverview: PropTypes.func,
}

export default function PlantFields({ plant, isLoaded, backToOverview }) {
    return (
        <BackgroundStyled>
            <PlantFieldsContainer>
                <ArrowDown data-testid="ArrowDown" onClick={backToOverview} />
                {isLoaded ? (
                    <>
                        <PlantThumbnail key={plant.id}>
                            <img src={plant.image_url} alt="" />
                            <h2>{plant.common_name}</h2>
                        </PlantThumbnail>
                        <FieldsWrapper>
                            <FieldsStyled>
                                <Field>Family</Field>
                                <Data>{plant.family}</Data>
                                <Field>Average-height (cm)</Field>
                                <Data>
                                    {plant.specifications.average_height.cm}
                                </Data>
                                <Field>Minimum-temperature (deg-c)</Field>
                                <Data>
                                    {plant.growth.minimum_temperature.deg_c}
                                </Data>
                                <Field>
                                    Soil texture <br /> (0)clay) - (10)rock
                                </Field>
                                <Data>{plant.growth.soil_texture}</Data>
                                <Field>Edible</Field>
                                <Data>{plant.edible ? 'yes' : 'no'}</Data>
                                <Field>Light (1-10)</Field>
                                <Data>{plant.growth.light}</Data>
                                <Field>Sowing</Field>
                                <Data>{plant.growth.sowing}</Data>
                                <Field>Row spacing</Field>
                                <Data>{plant.growth.row_spacing.cm}</Data>
                                <Field>Days to harvest</Field>
                                <Data>{plant.growth.days_to_harvest}</Data>
                            </FieldsStyled>
                            <Text>{plant.growth.description}</Text>
                        </FieldsWrapper>
                        <Link className="link" to="/spaceCreator">
                            <ButtonWrapper>
                                <ButtonStyled>Add to space</ButtonStyled>
                            </ButtonWrapper>
                        </Link>
                    </>
                ) : (
                    <LoadingMsgStyled />
                )}
            </PlantFieldsContainer>
        </BackgroundStyled>
    )
}

const ButtonWrapper = styled.div`
    display: flex;
    justify-content: center;
    margin-top: 5px;
`
const ButtonStyled = styled.button`
    color: var(--white);
    background-color: var(--darkGreen);
    border: none;
    border-radius: 5px;
    width: 100%;
    padding: 15px;
`

const BackgroundStyled = styled.div`
    background: var(--mediumGreen);
    width: 100%;
    height: 100%;
    padding: 20px;
`

const PlantFieldsContainer = styled.section`
    border-radius: 6px;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    padding: 20px;
    background-color: var(--white);
    width: 100%;
    overflow: none;
    margin-top: 50px;

    .link {
        color: var(--darkGrey);
        text-decoration: none;
    }
`
const ArrowDown = styled(ArrowIcon)`
    fill: var(--white);
    width: 20px;
    position: absolute;
    left: 20px;
    top: 20px;
`

const PlantThumbnail = styled.section`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 20px;

    img {
        height: 100px;
        width: 156px;
        object-fit: cover;
        border-radius: 5px;
        margin-bottom: 5px;
        background: var(--darkGrey);
        background-image: url(${logo});
        background-repeat: no-repeat;
    }
`

const FieldsWrapper = styled.div`
    max-height: 50vh;
    overflow-y: auto;
    border-top: 0.5px solid var(--lightGrey);
`

const FieldsStyled = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
`
const Data = styled.span`
    font-weight: 600;
    color: var(--darkGrey);
    line-height: 1.333;
    font-size: 75%;
    margin-top: 20px;
    margin-bottom: 10px;
`
const Field = styled(Data)`
    opacity: 50%;
`
const Text = styled.p`
    font-weight: 400;
    color: var(--darkGrey);
    line-height: 1.333;
    font-size: 75%;
    margin-top: 40px;
    margin-bottom: 10px;
`
const LoadingMsgStyled = styled.img`
    width: 40px;
    height: 40px;
    background-color: var(--mediumGreen);
    border-radius: 100%;
    animation: scaleout 1s infinite ease-in-out;

    @keyframes scaleout {
        0% {
            transform: scale(0);
        }
        100% {
            transform: scale(1);
            opacity: 0;
        }
    }
`
