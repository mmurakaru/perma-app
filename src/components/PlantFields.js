import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import logo from '../assets/perma_logo.svg'

PlantFields.propTypes = {
    plants: PropTypes.array,
}

export default function PlantFields({ plant }) {
    document.body.style.backgroundColor = '#AABB97'
    return (
        <PlantFieldsContainer>
            {console.log(plant)}
            <PlantThumbnail key={plant.id}>
                <img src={plant.image_url} alt="" />
                <h2>{plant.common_name}</h2>
            </PlantThumbnail>
            <FieldsStyled>
                <Field>Family</Field>
                <Data>{plant.family}</Data>
                <Field>Zone</Field>
                <Data>{(plant.native = '-')}</Data>
                <Field>Maximum-height</Field>
                <Data>{(plant.maximum_height = '-')}</Data>
                <Field>Minimum-temperature</Field>
                <Data>{(plant.minimum_temperature = '-')}</Data>
                <Field>Soil texture</Field>
                <Data>{(plant.soil_texture = '-')}</Data>
                <Field>Edible</Field>
                <Data>{(plant.edible = true ? 'Yes' : 'No')}</Data>
                <Field>Growth months</Field>
                <Data>{(plant.growth_months = '-')}</Data>
                <Field>Days to harvest</Field>
                <Data>{(plant.days_to_harvest = '-')}</Data>
                <Text>{plant.description}</Text>
            </FieldsStyled>
        </PlantFieldsContainer>
    )
}

const PlantFieldsContainer = styled.section`
    border-radius: 6px;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    padding: 20px;
    background-color: #f5f5f5;
    width: 100%;
    overflow: none;
    margin-top: 70px;
`

const PlantThumbnail = styled.section`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-bottom: 20px;

    img {
        background-image: url(${logo});
        height: 100px;
        width: 156px;
        object-fit: cover;
        border-radius: 5px;
        margin-bottom: 5px;
    }
`

const FieldsStyled = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
`

const Field = styled.span`
    font-weight: 600;
    color: #4a4a4a;
    opacity: 50%;
    line-height: 1.333;
    font-size: 75%;
    margin-top: 20px;
    margin-bottom: 10px;
`
const Data = styled.span`
    font-weight: 600;
    color: #4a4a4a;
    line-height: 1.333;
    font-size: 75%;
    margin-top: 20px;
    margin-bottom: 10px;
`
const Text = styled.p`
    font-weight: 400;
    color: #4a4a4a;
    opacity: 50%;
    line-height: 1.333;
    font-size: 75%;
    margin-top: 20px;
    margin-bottom: 10px;
`
