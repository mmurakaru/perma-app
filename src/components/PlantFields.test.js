import PlantFields from './PlantFields'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('PlantFields', () => {
    it('renders correctly', () => {
        const { container } = render(
            <PlantFields
                plant={{
                    id: 183086,
                    common_name: 'European mountain ash',
                    image_url:
                        'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                    family: 'Rosaceae',
                    specifications: {
                        average_height: {
                            cm: 300,
                        },
                    },
                    growth: {
                        description: 'Lorem ipsum',
                        sowing: 'Lorem ipsum',
                        days_to_harvest: 60,
                        row_spacing: {
                            cm: 30,
                        },
                        light: 7,
                        atmospheric_humidity: 5,
                        minimum_temperature: {
                            deg_c: -36,
                        },
                        soil_texture: 4,
                    },
                }}
                isLoaded={true}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    const onClickMock = jest.fn()

    it('calls onClick', () => {
        const { getByTestId } = render(
            <PlantFields backToOverview={onClickMock} />
        )
        userEvent.click(getByTestId('ArrowDown'), 'Coconut')
        expect(onClickMock).toHaveBeenCalled()
    })
})
