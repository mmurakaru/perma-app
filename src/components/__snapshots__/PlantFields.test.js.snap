// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantFields renders correctly 1`] = `
<div
  class="PlantFields__BackgroundStyled-sc-1k0boib-0 csTUyi"
>
  <section
    class="PlantFields__PlantFieldsContainer-sc-1k0boib-1 bYJscq"
  >
    <svg
      class="PlantFields__ArrowDown-sc-1k0boib-2 dA-DgKH"
      data-testid="ArrowDown"
    >
      arrow_down.svg
    </svg>
    <section
      class="PlantFields__PlantThumbnail-sc-1k0boib-3 iiNpVI"
    >
      <img
        alt=""
        src="https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed"
      />
      <h2>
        European mountain ash
      </h2>
    </section>
    <div
      class="PlantFields__FieldsWrapper-sc-1k0boib-4 dxAbef"
    >
      <div
        class="PlantFields__FieldsStyled-sc-1k0boib-5 dpQfph"
      >
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Family
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          Rosaceae
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Average-height (cm)
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          300
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Minimum-temperature (deg-c)
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          -36
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Soil texture 
          <br />
           (0)clay) - (10)rock
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          4
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Edible
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          no
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Light (1-10)
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          7
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Sowing
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          Lorem ipsum
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Row spacing
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          30
        </span>
        <span
          class="PlantFields__Field-sc-1k0boib-6 flvXCG"
        >
          Days to harvest
        </span>
        <span
          class="PlantFields__Data-sc-1k0boib-7 hGjkot"
        >
          60
        </span>
      </div>
      <p
        class="PlantFields__Text-sc-1k0boib-8 dPEROD"
      >
        Lorem ipsum
      </p>
    </div>
  </section>
</div>
`;
